### PR TITLE
Problem2: Word count

### PR DESCRIPTION
**Pull request by gsonal**

### CODE REVIEW

File: src/main/java/com/practise/tdd/WordCount.java [hunk 1]
```diff
@@ -1,20 +1,40 @@
...
 		
 		return output.substring(0, output.length()-2);
-	}	
+	}
+	
+	private Map<String, Integer> countUniqueWords(List<String> words) {
+		Map<String, Integer> uniqueWordCountMap = new HashMap<>();
+		words.forEach(word -> {
+			if(uniqueWordCountMap.containsKey(word)) {
+				Integer existingCount = uniqueWordCountMap.get(word);
```
Comment (suvendu123) [inline src/main/java/com/practise/tdd/WordCount.java]: you can avoid local Int variable : `uniqueWordCountMap.put(word, uniqueWordCountMap.get(word) +1);`  or like that.

Comment (gsonal) [reply src/main/java/com/practise/tdd/WordCount.java]: The parameter type I used is Integer and adding 1 without conversion will cause Autoboxing or Unboxing.

File: src/test/java/com/practise/tdd/WordCountShould.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+	WordCount wordCount;
+	
+	@Before
+	public void setup() {
+		wordCount = new WordCount();
+	}
+	
+	@Test
+	public void count_unique_words_in_single_word_string() {
```
Comment (premkumar-masilamani) [inline src/test/java/com/practise/tdd/WordCountShould.java]: Where is the test?

File: src/main/java/com/practise/tdd/WordCount.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package com.practise.tdd;
+
+public class WordCount {
+
```
Comment (premkumar-masilamani) [inline src/main/java/com/practise/tdd/WordCount.java:10]: What did work?. Why Green here?

File: src/test/java/com/practise/tdd/WordCountShould.java [hunk 4]
```diff
@@ -13,7 +14,7 @@ public void setup() {
 	}
 	
 	@Test
-	public void count_unique_words_in_single_word_string() {
-		
+	public void return_unique_words_in_single_word_string() {
```
Comment (premkumar-masilamani) [inline src/test/java/com/practise/tdd/WordCountShould.java]: This could be your starting point.

Comment (premkumar-masilamani) [reply src/test/java/com/practise/tdd/WordCountShould.java]: But the test case name is different from the asserts. Why do you need 1 appended here?

Comment (premkumar-masilamani) [reply src/test/java/com/practise/tdd/WordCountShould.java]: Could have used simple string and integer?

File: src/main/java/com/practise/tdd/WordCount.java [hunk 5]
```diff
@@ -3,7 +3,7 @@
 public class WordCount {
 
 	public String getUnqiueWordsFromString(String input) {
-		return input;
+		return input + " 1";
```
Comment (premkumar-masilamani) [inline src/main/java/com/practise/tdd/WordCount.java]: Not acceptable.

File: src/main/java/com/practise/tdd/WordCount.java [hunk 6]
```diff
@@ -1,9 +1,20 @@
 package com.practise.tdd;
 
+import java.util.Arrays;
+import java.util.List;
+
 public class WordCount {
 
-	public String getUnqiueWordsFromString(String input) {
-		return input + " 1";
+	public String getUnqiueWordsWithCountFromString(String input) {
```
Comment (premkumar-masilamani) [inline src/main/java/com/practise/tdd/WordCount.java]: `FromString` is un-necessary. The input parameters would reveal that anyway.

File: src/main/java/com/practise/tdd/WordCount.java [hunk 7]
```diff
@@ -1,9 +1,20 @@
...
+		String output = "";
+		
+		
+		List<String> uniqueWords = Arrays.asList(input.split(" "));		
+		for(String word : uniqueWords) {
+			output += word + " 1, ";
+		}
+		
+		return output.substring(0, output.length()-2);
```
Comment (premkumar-masilamani) [inline src/main/java/com/practise/tdd/WordCount.java]: Too primitive substring operations.

File: src/main/java/com/practise/tdd/WordCount.java (hunk 1 above)
Comment (premkumar-masilamani) [reply src/main/java/com/practise/tdd/WordCount.java]: Can you explain the `Autoboxing or Unboxing` issue?

File: src/test/java/com/practise/tdd/WordCountShould.java (hunk 2 above)
Comment (gsonal) [reply src/test/java/com/practise/tdd/WordCountShould.java]: As per course material, 'Not compiling' is a failure. So I already had a Red, which needed fixing, as first step.

File: src/main/java/com/practise/tdd/WordCount.java (hunk 3 above)
Comment (gsonal) [reply src/main/java/com/practise/tdd/WordCount.java:10]: Fixing compiling error was the Green step.

File: src/main/java/com/practise/tdd/WordCount.java (hunk 1 above)
Comment (gsonal) [reply src/main/java/com/practise/tdd/WordCount.java]: Adding the int value 1 to the returned Integer value from map.

File: src/main/java/com/practise/tdd/WordCount.java (hunk 5 above)
Comment (gsonal) [reply src/main/java/com/practise/tdd/WordCount.java]: Temp fix to pass the smallest test case. It wont be used as final solution.

File: src/test/java/com/practise/tdd/WordCountShould.java (hunk 2 above)
Comment (premkumar-masilamani) [reply src/test/java/com/practise/tdd/WordCountShould.java]: Please ignore that specific step from that video.

**Pull request merged**